File: tools/Legacy_GameRoot_CodeRED_Launchers_20260506/Code_RED_Launcher.cs
using System;
using System.Diagnostics;
using System.IO;

class CodeRedLauncher
{
    static int Main(string[] args)
    {
        string root = AppDomain.CurrentDomain.BaseDirectory;
        string bridgeDir = Path.Combine(root, "Code_RED_Launch");
        string target = "PlayRDR.exe";
        if (args.Length > 0 && string.Equals(args[0], "play", StringComparison.OrdinalIgnoreCase)) target = "PlayRDR.exe";
        if (args.Length > 0 && string.Equals(args[0], "direct", StringComparison.OrdinalIgnoreCase)) target = "RDR.exe";
        string targetPath = Path.Combine(root, target);
        if (!File.Exists(targetPath))
        {
            Console.WriteLine("Target executable not found: " + targetPath);
            return 1;
        }
        var psi = new ProcessStartInfo(targetPath)
        {
            WorkingDirectory = root,
            UseShellExecute = false,
        };
        psi.EnvironmentVariables["CODERED_BRIDGE_DIR"] = bridgeDir;
        psi.EnvironmentVariables["CODERED_ACTIVE_SESSION"] = Path.Combine(bridgeDir, "active_session.json");
        psi.EnvironmentVariables["CODERED_LAUNCH_PLAN"] = Path.Combine(bridgeDir, "launch_plan.json");
        psi.EnvironmentVariables["CODERED_HOOK_BOOTSTRAP"] = Path.Combine(root, "Code_RED_HookBridge", "hook_bootstrap.json");
        psi.EnvironmentVariables["CODERED_HOOK_PACK_DIR"] = Path.Combine(root, "Code_RED_HookBridge");
        Process.Start(psi);
        return 0;
    }
}

File: tools/Legacy_GameRoot_CodeRED_Launchers_20260506/Code_RED_Launcher.py
from __future__ import annotations
import os
import subprocess
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent
BRIDGE_DIR = ROOT / 'Code_RED_Launch'
TARGETS = {'primary': 'PlayRDR.exe', 'play': 'PlayRDR.exe', 'direct': 'RDR.exe'}
mode = sys.argv[1].strip().lower() if len(sys.argv) > 1 else 'primary'
target_name = TARGETS.get(mode, TARGETS['primary'])
target = ROOT / target_name
env = os.environ.copy()
env['CODERED_BRIDGE_DIR'] = str(BRIDGE_DIR)
env['CODERED_ACTIVE_SESSION'] = str(BRIDGE_DIR / 'active_session.json')
env['CODERED_LAUNCH_PLAN'] = str(BRIDGE_DIR / 'launch_plan.json')
env['CODERED_HOOK_BOOTSTRAP'] = str(ROOT / 'Code_RED_HookBridge' / 'hook_bootstrap.json')
env['CODERED_HOOK_PACK_DIR'] = str(ROOT / 'Code_RED_HookBridge')
runtime = BRIDGE_DIR / 'codered_bridge_runtime.py'
for candidate in (['py', '-3'], ['python']):
    if runtime.exists():
        try:
            subprocess.Popen(candidate + [str(runtime), str(BRIDGE_DIR)], cwd=str(ROOT), env=env)
            break
        except Exception:
            pass
if target.exists():
    subprocess.Popen([str(target)], cwd=str(ROOT), env=env)
else:
    print(f'Target executable not found: {target}')
